src/components/Insights.tsx

```diff
@@ -34,41 +35,74 @@ export default function Insights({ isAuthorized, data, updateEntryDate, selectio
     }
   }, [isAuthorized, router]);
 
-  // Fetch page data! (relies on `date`, only available on the client)
   useEffect(() => {
-    // Only run the side-effect ONCE per chosenDate
     if (loadOnce.current === chosenDate) {
       return;
     }
+
     loadOnce.current = chosenDate;
 
-    async function loadPageData() {
-      setLoading(true);
-      try {
-        const response = await getInsights(chosenDate);
-        if (response.insights) {
-          setInsights(response.insights);
-        }
-        else if (response.horoscope) {
-          setHoroscope(response.horoscope);
-        }
-        else {
-          setEmptyDay(true);
-        }
-        setLoading(false);
-      } catch (err) {
-        setError(true);
-        console.error(err);
```

Comment:
When we define this function outside of the `useEffect`, this function gets redefined every time the component is re-rendered. When we add `loadPageData` as a dependency, this causes the `useEffect` to get re-triggered in an infinite loop.

We *could* solve this using `useCallback`, however, a simpler solution is to just leave this function inside the `useEffect`. The only reason that it's a function is so that way we can use async await syntax. 

```typescript
useEffect(() => {
  async function whatever() {
    //we can use await in here, instead of having to use promise.then
  }
  whatever();
}, [])
```

Reply:
Gotcha I’ll move those back!